tests/components/lamarzocco/test_config_flow.py

```diff
@@ -435,6 +441,46 @@ async def test_bluetooth_discovery_errors(
     }
 
 
+async def test_dhcp_discovery(
+    hass: HomeAssistant,
+    mock_lamarzocco: MagicMock,
+    mock_cloud_client: MagicMock,
+    mock_device_info: LaMarzoccoDeviceInfo,
+) -> None:
+    """Test dhcp discovery."""
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={"source": SOURCE_DHCP},
+        data=DhcpServiceInfo(
+            ip="192.168.1.42",
+            hostname=mock_lamarzocco.serial_number,
```

Comment:
May be parametrize some hostnames to make sure all regex expressions are tested.

Reply:
I don't understand? I mean we are not evaluating the glob patterns here anyways, so we are just passing through the strings here.

Reply:
It's not a bad idea though (added it)